src/pages/OrderPopup.tsx

```diff
@@ -1,40 +1,108 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, FormEvent } from "react";
 import "@mantine/dates/styles.css";
 import "./OrderPopup.css";
 import { randomId, useDisclosure, useListState } from "@mantine/hooks";
-import {
-  Modal,
-  Button,
-  Text,
-  Title,
-  Avatar,
-  TextInput,
-  NumberInput,
-  ActionIcon,
-  Checkbox,
-} from "@mantine/core";
-import { DateInput } from "@mantine/dates";
-import OrderPopupQuantities from "./OrderPopupQuantities";
-
+import { Modal, Button, Title, NumberInput, Checkbox } from "@mantine/core";
+import Order from "./Order/OrderClass";
+import { useAuth } from "../AuthContext";
 
 const initialValues = [
   { label: "Approved", checked: false, key: randomId() },
   { label: "Opened", checked: false, key: randomId() },
   { label: "Viewed", checked: false, key: randomId() },
 ];
 
-const OrderPopup: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+const OrderPopup: React.FC<{ children: React.ReactNode; order: Order }> = ({
+  children,
+  order,
+}) => {
   const [opened, { open, close }] = useDisclosure(false);
   const [date, setDate] = useState<Date | null>(null);
   const [value, setValue] = useState<Date | null>(null);
-
+  const [numDiapers, setNumDiapers] = useState<number>(order.numDiapers);
   const [values, handlers] = useListState(initialValues);
 
+  const { logout, getUser, isStaff } = useAuth();
+  const user = getUser();
+
   const handleCheckboxChange = (index: any) => {
     handlers.setState((current) =>
       current.map((value, i) => ({ ...value, checked: i === index }))
     );
   };
+  const [sizes, setSizes] = useState<number[]>(order.diaperDist);
+  const { currentUser } = useAuth();
+
+  const handleChange = (val: number, index: number) => {
+    const newSizes = sizes.map((num, i) => {
+      if (index === i) {
```

Comment:
This is fine, but remember you can always use the ternary operator for if statements like these:
eg. return (index ===i) ? val : num